### ISSUE TITLE
learning rate can't update in resuming training

### ISSUE THREAD

**Issue by TLedTS** [horovod/horovod#3370]
**Environment:**
1. Framework: PyTorch_lighting
2. Framework version:1.5.4 with pytorch 1.10.0
3. Horovod version:0.23.0
4. MPI version:3.0.0
5. CUDA version:11.1
6. NCCL version:2.7.8
7. Python version:3.7.11
8. Spark / PySpark version:
9. Ray version:
10. OS and version:
11. GCC version:
12. CMake version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe erroneous behavior you're observing and steps to reproduce it.
bug:
When i resume trained the model with the LearningRateMonitor, then the lr written in the log won't update anymore but just keep the same with original status from resume ckpt file. 

Use the MNIST as an example, train it from scratch and save a model in middle.
Then pass the model path to trainer to resume train the model.
Now check the log in tensorboard, you can see the lr-Adam won't update.
But for the 'lr' we write in log , it is normal.
And it will be normal when the strategy is 'ddp' or with the version 1.4.2 of pytorch_lightning.

<img width="382" alt="Screen Shot 2022-01-17 at 17 58 04" src="https://user-images.githubusercontent.com/27993525/149748257-076d8d31-8013-4287-9d20-1db0d1ffa9a3.png">
If anyone has any suggestion, please kindly let me know! Thanks a lot!

code:
```
import os
import torch
from torch import nn
import torch.nn.functional as F
from torchvision.datasets import MNIST
from torch.utils.data import DataLoader, random_split
from torchvision import transforms
import pytorch_lightning as pl
from pytorch_lightning.callbacks import LearningRateMonitor
import horovod.torch as hvd

class LitAutoEncoder(pl.LightningModule):
    def __init__(self):
        super().__init__()
        self.encoder = nn.Sequential(nn.Linear(28 * 28, 128), nn.ReLU(), nn.Linear(128, 3))
        self.decoder = nn.Sequential(nn.Linear(3, 128), nn.ReLU(), nn.Linear(128, 28 * 28))

    def forward(self, x):
        # in lightning, forward defines the prediction/inference actions
        embedding = self.encoder(x)
        return embedding

    def training_step(self, batch, batch_idx):
        # training_step defines the train loop. It is independent of forward
        x, y = batch
        x = x.view(x.size(0), -1)
        z = self.encoder(x)
        x_hat = self.decoder(z)
        loss = F.mse_loss(x_hat, x)
        self.log("train_loss", loss)
        self.log("lr", self.optimizer.param_groups[0]['lr'])
        return loss

    def configure_optimizers(self):
        self.optimizer = torch.optim.Adam(self.parameters(), lr=0.01)
        lr_scheduler = torch.optim.lr_scheduler.CosineAnnealingWarmRestarts(optimizer=self.optimizer, T_0=10, T_mult=2, eta_min=0)
        return [self.optimizer], [lr_scheduler]


def main():
    dataset = MNIST(os.getcwd(), download=True, transform=transforms.ToTensor())
    train, val = random_split(dataset, [55000, 5000])

    autoencoder = LitAutoEncoder()
    lr_monitor = LearningRateMonitor(logging_interval='step')
    trainer = pl.Trainer(
        gpus=1,
        max_epochs=64,
        strategy='horovod',
        callbacks=[lr_monitor],
        limit_train_batches=32
        )
    trainer.fit(autoencoder, DataLoader(train), DataLoader(val), ckpt_path='lightning_logs/version_14/checkpoints/epoch=4-step=63.ckpt')


if __name__ == "__main__":
    main()
```
